lightblue-client-integration-test/src/test/java/com/redhat/lightblue/client/integration/test/LockingTest.java

```diff
@@ -32,13 +36,43 @@ public void testAcquireAndRelease() throws Exception {
         assertTrue(lbLocker.release("test-lock"));
     }
 
+    @Test(expected = LightblueResponseException.class)
+    public void testPingWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.ping("test-lock");
+    }
+
+    @Test(expected = LightblueResponseException.class)
+    public void testCountWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.getLockCount("test-lock");
+    }
+
+    @Test(expected = LightblueResponseException.class)
+    public void testReleaseWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.release("test-lock");
+    }
+
     @Test
     public void testLock() throws LightblueException, IOException {
         Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
         try (Lock lock = lbLocker.lock("test-lock")) {
             assertEquals(1, lock.getLockCount());
         }
-        assertEquals(0, lbLocker.getLockCount("test-lock"));
+
+        //assert the lock was successfully released.
+        try {
+            lbLocker.getLockCount("test-lock");
+            fail();
+        } catch (LightblueResponseException e) {
+            Error error = e.getLightblueResponse().getLightblueErrors()[0];
+            assertNotNull(error);
+            assertTrue(error.getMsg().contains("InvalidLockException: test-lock"));
+        }
```

Comment:
Hm, it is a bit misleading I think, but reasonable. However, the convention
for all the locking APIs is that if the lock is invalid, and a lock becomes
invalid once it is unlocked, they throw exceptions. If it is changed here,
it should be changed in others as well.

One way to deal with ugly code is to create separate lock exceptions for
client, and to make them runtime exceptions.

On Wed, Jan 13, 2016 at 7:59 AM, Dennis Crissman <notifications@github.com>
wrote:

> In
> lightblue-client-integration-test/src/test/java/com/redhat/lightblue/client/integration/test/LockingTest.java
> <https://github.com/lightblue-platform/lightblue-client/pull/234#discussion_r49599850>
> :
>
> >      @Test
> >      public void testLock() throws LightblueException, IOException {
> >          Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
> >          try (Lock lock = lbLocker.lock("test-lock")) {
> >              assertEquals(1, lock.getLockCount());
> >          }
> > -        assertEquals(0, lbLocker.getLockCount("test-lock"));
> > +
> > +        //assert the lock was successfully released.
> > +        try {
> > +            lbLocker.getLockCount("test-lock");
> > +            fail();
> > +        } catch (LightblueResponseException e) {
> > +            Error error = e.getLightblueResponse().getLightblueErrors()[0];
> > +            assertNotNull(error);
> > +            assertTrue(error.getMsg().contains("InvalidLockException: test-lock"));
> > +        }
>
> @bserdar <https://github.com/bserdar> - Should a get lock count return an
> error from lightblue if the resourceId is unknown? Or could it just return
> 0?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/234/files#r49599850>
> .
>